Smime_sign_regional_characters

ckaadic opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Currently the results of smime-sign-email is not valid for regional special characters like ä, Ø, ß and others. Fixed this by using transport encoding.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

adi88d reviewed (commented):
Hi @ckaadic,
Thank you for your contribution!
can you please tell me how to reproduce the issue with `mime-sign-email` command?
i tried to run it with the special characters and it works with no erros
<img width="729" alt="image" src="https://user-images.githubusercontent.com/46249224/210788341-22d3343c-0dff-4425-bee7-422fe312ec5c.png">



and Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for demo over DFIR.

ckaadic commented:
Hello @adi88d. My initial description was missleading so I adjusted it: Currently the results of smime-sign-email is not valid for regional special characters like ä, Ø, ß and others. Fixed this by using transport encoding.

This means yes, smime-sign-email will run without showing an error. But mail gateways will not accept the smime and will not deliver the mail. This can easily be resolved by using transport encoding which I did. With a second thought it might be enough (for me) to have transport encoding as an optional argument (if someone needs it without). So I can change it to support both with an argument, if you like that more.

ckaadic commented:
The send-mail command will run into a timeout with special character and will return "Mail sent successfully" when run without special character:
![image](https://user-images.githubusercontent.com/48683125/210802661-8b6e4e3f-4e17-4a3f-b822-d459f94bfe89.png)

adi88d commented:
Thanks @ckaadic for your response!
please add to the command the `use_transport_encoding` and change the py and yml files.

adi88d commented on Packs/SMIME_Messaging/Integrations/SMIME_Messaging/SMIME_Messaging.py:
```diff
@@ -9,8 +9,9 @@
+import quopri
 import warnings
-warnings.simplefilter("default")
+#warnings.simplefilter("default")
```
revert this line

adi88d commented on Packs/SMIME_Messaging/Integrations/SMIME_Messaging/SMIME_Messaging.py:
```diff
@@ -41,17 +42,30 @@ def sign_email(client: Client, args: Dict):
     """
```
please add the `use_transport_encoding: bool` arg to `sign_email` method
and update the code by checking this arg once:

> if use_transport_encoding:
     your code part
  else:
      the old code
      
in the `main` method add:
`use_transport_encoding: bool = argToBoolean(demisto.params().get('use_transport_encoding'))`
and pass it to `sign_email` method

ckaadic replied:
use_transport_encoding is not a parameter at the moment, but an argument, as you might want to change this depending on the message. I set it in the beginning of the function and use the boolean for the check afterwards.

adi88d replied:
@ckaadic you right its the command parameter, my bad.
what i meant in the last comment is to update the code by checking `use_transport_encoding` arg once inside `sign_email`, you check it 3 times, it should be:

> if use_transport_encoding:
> your code part
> else:
> the old code

the old part part should start from the line:
> message_body = args.get('message_body', '')
to:
> client.smime.write(out, p7, buf, SMIME.PKCS7_TEXT)

adi88d commented on Packs/SMIME_Messaging/ReleaseNotes/1_0_7.md:
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### SMIME Messaging
+- Fixed an issue where regional characters in **message_body** lead to errors when running ***smime-sign-email*** 
```
```suggestion
- Added the **use_transport_encoding** argument to the ***smime-sign-email*** command.
```

adi88d reviewed (changes requested):
Thanks @ckaadic 
please resolve the conflicts and my comments below
and we can set up the demo at Wednesday at 12pm CET
let me know if that works for you.

ckaadic commented:
12pm works for me. Changed two parts as asked and modified the third, as this was possible at the moment

adi88d commented:
please resolve the branch conflicts

ckaadic commented:
changed the if part if the sign_email function as requested. Resolved conflicts and bumped the version of the Pack, as required

ckaadic commented:
rebased as the "circleci: Setup Environment" step was asking for it. Still fails, but I can't find the issue. Can you give me a hint

adi88d reviewed (approved):
great!
